tests/ssl/Makefile

```diff
@@ -0,0 +1,233 @@
+SUBJ_C ?= GB
+SUBJ_ST ?= London
+SUBJ_L ?= London
+SUBJ_O ?= Global Security
+SUBJ_OU ?= IT
+SUBJ_CN ?= Test certificate
+SUBJECT ?= /C=$(SUBJ_C)/ST=$(SUBJ_ST)/L=$(SUBJ_L)/O=$(SUBJ_O)/OU=$(SUBJ_OU)/CN=$(SUBJ_CN)
+EXPIRE_DAYS ?= 18250
+
+# $1 => group-name
+# $2 => name
+# $3 => key size in bits (leave empty for default, 2048)
+define GenerateRSAKey
+$(1)/$(2)/$(2).key:
+	@mkdir -p $$(shell dirname "$$@")
+	@echo -e "\e[1;37m==> Generating private key $(1)/$(2)/$(2).key\e[0m"
+	[ -e "$(1)/$(2)/$(2).key" ] && echo "SKIP" || \
+	openssl genrsa -out "$(1)/$(2)/$(2).key" $(if $(3),$(3),2048) && \
+	chmod 400 "$(1)/$(2)/$(2).key"
```

Comment:
Does it _**need**_ to be a makefile? I mean, at this point, a Perl script might be more readable and use fewer `$`. How about just a Bash script to regenerate everything? It's not like we need to track dependencies too closely here.